session.go

```diff
@@ -298,6 +298,32 @@ func newSenderForSession(ctx context.Context, s *Session, target string, opts *S
 	return l, nil
 }
 
+// NewTransactionController creates a transaction controller, which can declare
+// and discharge transactions.
+func (s *Session) NewTransactionController(ctx context.Context, opts *TransactionControllerOptions) (*TransactionController, error) {
+	return newTransactionControllerForSession(ctx, s, opts, senderTestHooks{})
+}
+
+func newTransactionControllerForSession(ctx context.Context, s *Session, opts *TransactionControllerOptions, hooks senderTestHooks) (*TransactionController, error) {
+	if opts == nil {
+		opts = &TransactionControllerOptions{}
+	}
+
+	sender, err := newTransactionControllerSender(s, opts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := sender.attach(ctx); err != nil {
```

Comment:
I think this will need some retooling.  Per [4.5.1](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#type-coordinator) the controller is required to verify that the coordinator meets the requirements.  e.g. if you ask for `multi-ssns-per-txn` and the coordinator responds with something else then the control link should be closed.

Reply:
Agreed, I'll take a look at this. I think I also need to expose the returned features as well, but I need to check.